crates/rolldown/src/utils/chunk/render_chunk_exports.rs

```diff
@@ -23,19 +23,30 @@ pub fn render_chunk_exports(
   match options.format {
     OutputFormat::Esm => {
       let mut s = String::new();
+      let mut export_default_count = 0;
       let rendered_items = export_items
         .into_iter()
         .map(|(exported_name, export_ref)| {
           let canonical_ref = link_output.symbols.par_canonical_ref_for(export_ref);
           let symbol = link_output.symbols.get(canonical_ref);
-          let canonical_name = &chunk.canonical_names[&canonical_ref];
+
+          let canonical_name;
           if let Some(ns_alias) = &symbol.namespace_alias {
             let canonical_ns_name = &chunk.canonical_names[&ns_alias.namespace_ref];
+            canonical_name = if export_default_count == 0 {
```

Comment:
if you directly use `            canonical_name = chunk.canonical_names[&ns_alias.namespace_ref].clone();
`, 
we use `issue/2085` as a example, the `a~1` would be 
```js
## a~1.mjs

```js
import { import_b } from "./a.mjs";

var import_b = import_b._default;
export { import_b as default };
```